frontend/src/pages/PipelineDetails.tsx

```diff
@@ -206,35 +206,7 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
 
         // V1: Convert the run's pipeline spec to YAML to be displayed as the pipeline's source.
         // V2: Use the pipeline spec string directly because it can be translated in JSON format.
-        try {
-          const workflowManifestString = RunUtils.getWorkflowManifest(runDetails.run) || '';
-          const workflowManifest = JSON.parse(workflowManifestString || '{}');
-          try {
-            if (WorkflowUtils.isPipelineSpec(workflowManifestString)) {
-              templateString = workflowManifestString;
-            } else {
-              templateString = JsYaml.safeDump(workflowManifest);
-            }
-          } catch (err) {
-            await this.showPageError(
-              `Failed to parse pipeline spec from run with ID: ${runDetails.run!.id}.`,
-              err,
-            );
-            logger.error(
-              `Failed to convert pipeline spec YAML from run with ID: ${runDetails.run!.id}.`,
-              err,
-            );
-          }
-        } catch (err) {
-          await this.showPageError(
-            `Failed to parse pipeline spec from run with ID: ${runDetails.run!.id}.`,
-            err,
-          );
-          logger.error(
-            `Failed to parse pipeline spec JSON from run with ID: ${runDetails.run!.id}.`,
-            err,
-          );
-        }
```

Comment:
We cannot delete the logic here. This is for KFPv1 pipeline. You can attempt to read `pipeline_manifest` if the V1 logic failed to find a templateString. The high level idea is:

1. If `pipeline_manifest` exists and `isFeatureEnabled(FeatureKey.V2_ALPHA)` is true, read `pipeline_manifest` for templateString.
2. Otherwise, read `workflow_manifest` for templateString which is the existing logic.